src/citrine/informatics/generative_design.py

```diff
@@ -128,9 +136,11 @@ def __init__(
         min_fingerprint_similarity: float,
         mutation_per_seed: int,
         structure_exclusions: Optional[List[StructureExclusion]],
+        min_substructure_counts: Dict[str, int],
```

Comment:
@kylecsnow Do customers always need to provide `min_substructure_counts`? If not, let's have it default to `{}` for convenience.

Reply:
@anoto-moniz This should be an optional argument generally.  But I've got the same question about this as in my other comment reply, should this instead be `min_substructure_counts: Optional[Dict[str, int]],`? 
(please see other comment thread)